### PR TITLE
Add staging workflow dispatch

### PR DESCRIPTION
**Pull request by ewlu**

### CODE REVIEW

File: .github/workflows/staging.yaml [hunk 1]
```diff
@@ -4,10 +4,42 @@ on:
...
+        description: 'Patch Timestamp (UTC)'
+        required: true
+      baseline_gcc_hash:
+        description: 'Baseline hash'
+        required: false
+      tip_of_tree_hash:
+        description: 'Trunk hash'
+        required: false
+      patch_id:
+        description: 'Patch id (patch series not supported)'
```
Comment (patrick-rivos) [inline .github/workflows/staging.yaml:21]:
```suggestion
        description: 'Patch id'
```

File: .github/workflows/staging.yaml [hunk 2]
```diff
@@ -4,10 +4,42 @@ on:
...
+        required: false
+      original_patch_issue_num:
+        description: 'Issue number to link against'
+        required: false
 
 jobs:
+  run-workflow-dispatch:
```
Comment (patrick-rivos) [inline .github/workflows/staging.yaml:28]:
```suggestion
  manually-triggered-workflow-dispatch:
```

**Review by patrick-rivos — APPROVED** [review]
LGTM. Out of curiosity - what happens if we pass in a baseline hash that doesn't exist?

**Comment by ewlu** [comment]
I believe `init-submodules` will fail to checkout the baseline GCC hash

**Comment by patrick-rivos** [comment]
Ah I meant one that doesn't have a corresponding postcommit run.

**Comment by ewlu** [comment]
If that's the case, I think it'll try to apply as usual. It'll only break once it gets to downloading the comparison artifacts so it'll terminate at downloading the postcommit sum files or it will report no baseline. I can't remember if downloading the sum files is marked `continue-on-error: true`

**Pull request merged**